docs/core/testing/unit-testing-code-coverage.md

```diff
@@ -0,0 +1,406 @@
+          "Lines": {
+            "8": 11,
+            "9": 11,
+            "10": 3,
+            "11": 3,
+            "14": 22,
+            "15": 7,
+            "16": 7,
+            "17": 4,
+            "18": 4,
+            "20": 3,
+            "21": 4,
+            "23": 11
+          },
+          "Branches": [
+            {
+              "Line": 9,
+              "Offset": 7,
+              "EndOffset": 9,
+              "Path": 0,
+              "Ordinal": 0,
+              "Hits": 3
+            },
+            {
+              "Line": 9,
+              "Offset": 7,
+              "EndOffset": 14,
+              "Path": 1,
+              "Ordinal": 1,
+              "Hits": 8
+            },
+            {
+              "Line": 14,
+              "Offset": 57,
+              "EndOffset": 18,
+              "Path": 1,
+              "Ordinal": 5,
+              "Hits": 7
+            },
+            {
+              "Line": 16,
+              "Offset": 27,
+              "EndOffset": 29,
+              "Path": 0,
+              "Ordinal": 2,
+              "Hits": 4
+            },
+            {
+              "Line": 16,
+              "Offset": 27,
+              "EndOffset": 34,
+              "Path": 1,
+              "Ordinal": 3,
+              "Hits": 3
+            },
+            {
+              "Line": 14,
+              "Offset": 57,
+              "EndOffset": 59,
+              "Path": 0,
+              "Ordinal": 4,
+              "Hits": 4
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+> [!NOTE]
+> There is a subtle difference between the outputs of the two `dotnet test` commands. The first output was an XML file, whereas the last is a JSON file.
+
+## Use with .NET global tooling
+
+To use coverlet globally, it can be installed as a [.NET global tool](../tools/global-tools.md). Optionally, run the following [`dotnet tool install`](../tools/dotnet-tool-install.md) command:
+
+```dotnetcli
+dotnet tool install --global coverlet.console
+```
+
+The [coverlet.console](https://www.nuget.org/packages/coverlet.console) NuGet package is added to the environment globally. Now, `coverlet` can be used as a CLI command:
+
+```console
+coverlet Full\Path\To\TestProject.dll --target "dotnet" --targetargs "test . --no-build"
+```
+
+## Generate reports
+
+Now that you're able to collect data from unit test runs, you can generate reports using [ReportGenerator](https://github.com/danielpalme/ReportGenerator). To install the [ReportGenerator](https://www.nuget.org/packages/dotnet-reportgenerator-globaltool) NuGet package as a [.NET global tool](../tools/global-tools.md), use the [`dotnet tool install`](../tools/dotnet-tool-install.md) command:
+
+```dotnetcli
+dotnet tool install -g dotnet-reportgenerator-globaltool
+```
+
+Run the tool and provide the desired options, given the output *coverage.cobertura.xml* file from the previous test run.
```

Comment:
What about the JSON file?

Reply:
Ironically, as it turns out... ReportGenerator doesn't take this format in. This file is actually consumed by Visual Studio I believe. What should I say about it?

Reply:
Link to VS docs that show how to use it, and warn up front so they can decide against using this option if they aren't using VS